src/auth.py

```diff
@@ -0,0 +1,134 @@
+import base64
+import hashlib
+import requests
+import random
+import string
+import datetime
+import urllib
+
+from requests_toolbelt.multipart.encoder import MultipartEncoder
+
+from bb2 import Bb2
+
+BB2_AUTH_URL = "{}/v{}/o/authorize"
+BB2_TOKEN_URL = "{}/v{}/o/token/"
+
+
+class AuthRequest:
+
+    def __init__(self, bb: Bb2):
+        self.bb = bb
+        self.auth_base_url = BB2_AUTH_URL.format(bb.get_config()['base_url'], bb.get_config()['version'])
```

Comment:
I know it's probably tricky since Dave is working on the PR now but I think this can be referenced just on the class itself instead of from the config.
```suggestion
        self.auth_base_url = BB2_AUTH_URL.format(bb.base_url, bb.version)
```